File: app/src/main/java/com/estrrado/tradopay/ui/modules/home/HomeListItemDetailFragment.java
package com.estrrado.tradopay.ui.modules.home;

import android.content.Intent;
import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.estrrado.tradopay.ui.modules.OnItemSelectedlistner;
import com.estrrado.tradopay.ui.modules.home.adapter.HomeDetailedrRoundItemAdapter;
import com.estrrado.tradopay.ui.modules.home.adapter.HomeItemsDetailAdapter;
import com.estrrado.tradopay.ui.modules.home.adapter.HomeItemsListAdapter;
import com.example.tradopay.R;

/**
 * A simple {@link Fragment} subclass.
 * Use the {@link HomeListItemDetailFragment#newInstance} factory method to
 * create an instance of this fragment.
 */
public class HomeListItemDetailFragment extends Fragment {

    // TODO: Rename parameter arguments, choose names that match
    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
    private static final String ARG_PARAM1 = "param1";
    private static final String ARG_PARAM2 = "param2";

    // TODO: Rename and change types of parameters
    private String mParam1;
    private String mParam2;

    public HomeListItemDetailFragment() {
        // Required empty public constructor
    }

    /**
     * Use this factory method to create a new instance of
     * this fragment using the provided parameters.
     *
     * @param param1 Parameter 1.
     * @param param2 Parameter 2.
     * @return A new instance of fragment HomeItemDetailFragment.
     */
    // TODO: Rename and change types and number of parameters
    public static HomeListItemDetailFragment newInstance(String param1, String param2) {
        HomeListItemDetailFragment fragment = new HomeListItemDetailFragment();
        Bundle args = new Bundle();
        args.putString(ARG_PARAM1, param1);
        args.putString(ARG_PARAM2, param2);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mParam1 = getArguments().getString(ARG_PARAM1);
            mParam2 = getArguments().getString(ARG_PARAM2);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view=inflater.inflate(R.layout.fragment_home_detail, container, false);
        RecyclerView rcvItems= view.findViewById(R.id.rcvItems);
        RecyclerView rcvHomeItems= view.findViewById(R.id.rcvHomeItemsDetail);
        rcvItems.setLayoutManager(new GridLayoutManager(getActivity(),4));
        rcvHomeItems.setLayoutManager(new LinearLayoutManager(getActivity()));
        rcvItems.setAdapter(new HomeDetailedrRoundItemAdapter(new OnItemSelectedlistner() {
            @Override
            public void onItemClick(Object object, int position) {
               // Toast.makeText(getContext(),"Clicked",Toast.LENGTH_LONG).show();
               //getActivity().getSupportFragmentManager().beginTransaction().replace(R.id.nav_host_fragment,new HomeItemDetailChildFragment()).commit();
                startActivity(new Intent(getActivity(),HomeDetailedActivity.class));
            }
        }));
        rcvHomeItems.setAdapter(new HomeItemsListAdapter(new OnItemSelectedlistner() {
            @Override
            public void onItemClick(Object object, int position) {
               // Toast.makeText(getContext(),"Clicked",Toast.LENGTH_LONG).show();
               // getActivity().getSupportFragmentManager().beginTransaction().replace(R.id.nav_host_fragment,new HomeItemDetailChildFragment()).commit();
            }
        }));
        return view;
    }
}

File: app/src/main/java/com/estrrado/tradopay/fragments/HomeListItemDetailFragment.kt
package com.estrrado.tradopay.fragments

import com.example.tradopay.R
import android.os.Bundle
import com.estrrado.tradopay.fragments.HomeListItemDetailFragment
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.LinearLayoutManager
import com.estrrado.tradopay.helpers.ItemOffsetDecoration
import com.estrrado.tradopay.adapter.HomeDetailedrRoundItemAdapter
import com.estrrado.tradopay.listener.OnItemSelectedlistner
import android.content.Intent
import android.view.View
import androidx.fragment.app.Fragment
import com.estrrado.tradopay.activity.HomeDetailedActivity
import com.estrrado.tradopay.adapter.HomeItemsListAdapter
import kotlinx.android.synthetic.main.fragment_home_item_list.*

/**
 * A simple [Fragment] subclass.
 * Use the [HomeListItemDetailFragment.newInstance] factory method to
 * create an instance of this fragment.
 */
class HomeListItemDetailFragment : BaseFragment() {
    // TODO: Rename and change types of parameters
    private var mParam1: String? = null
    private var mParam2: String? = null
    var images_detail = intArrayOf(R.drawable.rest, R.drawable.bak, R.drawable.groc,
            R.drawable.frut, R.drawable.book, R.drawable.cloth, R.drawable.elec, R.drawable.medi)
    var headings_detail = arrayOf("Restaurants", "Bakery", "Groceries", "Fruits &\nVegetables",
            "Books", "Cloths", "Electronics", "Medicines")
    var headings = arrayOf(arrayOf("Delivery At Your Doorsteps", "Restaurants,Bakery,Grocery," +
            "Household Items and many other items", "Order Now"), arrayOf("Book Your Appointments",
            "Doctors,Car/Bike Wash & Service centers,, Saloons,Spa/Parlors,Gyms and other services",
            "Book Now"), arrayOf("Cashbacks and Offers", "View Cashbacks and Offers", "View All"))
    var images = intArrayOf(R.drawable.scty_grl, R.drawable.appoinmt, R.drawable.sale)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
          /*  mParam1 = arguments!!.getString(ARG_PARAM1)
            mParam2 = arguments!!.getString(ARG_PARAM2)*/
        }
    }

    override fun initLayout(inflater: LayoutInflater, container: ViewGroup?): View {
        val view = inflater.inflate(R.layout.fragment_home_detail, container, false)
        val rcvItems: RecyclerView = view.findViewById(R.id.rcvItems)
        val rcvHomeItems: RecyclerView = view.findViewById(R.id.rcvHomeItemsDetail)
        rcvItems.layoutManager = GridLayoutManager(activity, 4)
        rcvHomeItems.layoutManager = LinearLayoutManager(activity)
        rcvItems.addItemDecoration(ItemOffsetDecoration(2))
        rcvItems.adapter = HomeDetailedrRoundItemAdapter(object : OnItemSelectedlistner {
            override fun onItemClick(`object`: Any?, position: Int) {
                startActivity(Intent(activity, HomeDetailedActivity::class.java))
            }
        }, headings_detail, images_detail)

        return view
    }

    override fun initListeners(view: View) {
        rcvHomeItems.adapter = HomeItemsListAdapter(object : OnItemSelectedlistner {
            override fun onItemClick(`object`: Any?, position: Int) {
                // Toast.makeText(getContext(),"Clicked",Toast.LENGTH_LONG).show();
                // getActivity().getSupportFragmentManager().beginTransaction().replace(R.id.nav_host_fragment,new HomeItemDetailChildFragment()).commit();
            }
        }, headings, images)
    }

    override fun initObservers() {
        TODO("Not yet implemented")
    }

    companion object {
        // TODO: Rename parameter arguments, choose names that match
        // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
        private const val ARG_PARAM1 = "param1"
        private const val ARG_PARAM2 = "param2"

        /**
         * Use this factory method to create a new instance of
         * this fragment using the provided parameters.
         *
         * @param param1 Parameter 1.
         * @param param2 Parameter 2.
         * @return A new instance of fragment HomeItemDetailFragment.
         */
        // TODO: Rename and change types and number of parameters
        fun newInstance(param1: String?, param2: String?): HomeListItemDetailFragment {
            val fragment = HomeListItemDetailFragment()
            val args = Bundle()
            args.putString(ARG_PARAM1, param1)
            args.putString(ARG_PARAM2, param2)
            fragment.arguments = args
            return fragment
        }
    }
}